src/sqlfluff/rules/L035.py

```diff
@@ -46,26 +46,24 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
         5.b. We reach the end of case when without matching "NULL": the rule passes
         """
         if context.segment.is_type("case_expression"):
-            fixes: List[LintFix] = []
             for idx, seg in enumerate(context.segment.segments):
-                # When we find ELSE we delete
-                # everything up to NULL
-                if fixes:
-                    fixes.append(LintFix.delete(seg))
-                    # Safe to look for NULL, as an expression
-                    # would contain NULL but not be == NULL
-                    if seg.raw_upper == "NULL":
-                        return LintResult(anchor=context.segment, fixes=fixes)
-
-                if not fixes and seg.name == "else":
-                    fixes.append(LintFix.delete(seg))
-                    # Walk back to remove indents/whitespaces
+                # When we find ELSE with NULL, we delete the whole else clause.
+                # Here, it's safe to look for NULL, as an expression would
+                # *contain* NULL but not be == NULL.
+                if seg.is_type("else_clause") and any(
+                    child.raw_upper == "NULL" for child in seg.segments
+                ):
```

Comment:
I'm a little worried about this. So any direct child that has `NULL` in it will be deleted?

What about this:

```sql
    select
        case name
            when 'cat' then true
            when 'dog' then true
            else name is null
        end
    from x
```

To me the L035 test cases look light and could do with adding the above one anyway. At the moment it has the following test cases:
- one without null
- one with null with no column
- one with null with column
- one with null in complex nested `else iff` statement

Reply:
The test above passes (no lint error, that is). That's because `name is null` is parsed as an `expression`, and the `null` is a child segment of the expression, hence the `any()` doesn't trigger because this is what the loop is looking at:
```
(Pdb) pp [child.raw_upper for child in seg.segments]
['ELSE', '', ' ', 'NAME IS NULL', '']
```

Reply:
Good stuff. Thanks for checking.